src/docs/json_rpc.md

```diff
@@ -208,138 +177,98 @@ the frontend.
+        const taskWatcher = ctx.container.get(TaskWatcher);
+        return connection.createProxy<TaskServer>(taskPath, taskWatcher.getTaskClient());
     }).inSingletonScope();
 
 ```
 
 Let's go line by line:
 
 ``` typescript
-        const loggerWatcher = ctx.container.get(LoggerWatcher);
+        const connection = ctx.container.get(WebSocketConnectionProvider);
 ```
 
-Here we're creating a watcher, this is used to get notified about events
-from the backend by using the loggerWatcher client
-(loggerWatcher.getLoggerClient())
-
-See more information about how events work in theia [here](/docs/Events#events).
+Here we're getting the websocket connection, this will be used to create a proxy from.
 
 ``` typescript
-        const connection = ctx.container.get(WebSocketConnectionProvider);
+        const taskWatcher = ctx.container.get(TaskWatcher);
 ```
 
-Here we're getting the websocket connection, this will be used to create a proxy from.
+Here we're creating a watcher, this is used to get notified about events
+from the backend by using the `taskWatcher`'s client
+(`taskWatcher.getTaskClient()`)
+
+See more information about how events work in theia [here](/docs/Events#events).
 
 ``` typescript
-        return connection.createProxy<ILoggerServer>("/services/logger", loggerWatcher.getLoggerClient());
+        return connection.createProxy<TaskServer>(taskPath, taskWatcher.getTaskClient());
 ```
 
-As the second argument, we pass a local object to handle JSON-RPC messages from the remote object.
+As the second argument, we pass a local object to handle RPC messages from the remote object.
 Sometimes the local object depends on the proxy and cannot be instantiated before the proxy is instantiated.
-In such cases, the proxy interface should implement `JsonRpcServer` and the local object should be provided as a client.
+In such cases, the proxy interface should implement `RpcServer` and the local object should be provided as a client.
 
 ```ts
-export type JsonRpcServer<Client> = Disposable & {
+export type RpcServer<Client> = Disposable & {
+    /**
+     * If this server is a proxy to a remote server then
+     * a client is used as a local object
+     * to handle RPC messages from the remote server.
+     */
     setClient(client: Client | undefined): void;
+    getClient?(): Client | undefined;
 };
 
-export interface ILoggerServer extends JsonRpcServery<ILoggerClient> {
+export interface TaskServer extends RpcServer<TaskClient>  {
     // ...
 }
 
-const serverProxy = connection.createProxy<ILoggerServer>("/services/logger");
-const client = loggerWatcher.getLoggerClient();
+const serverProxy = connection.createProxy<TaskServer>("/services/task");
+const client = taskWatcher.getTaskClient();
 serverProxy.setClient(client);
 ```
 
-So here at the last line we're binding the ILoggerServer interface to a
-JsonRpc proxy.
-
-Note that his under the hood calls:
```

Comment:
Again, do you consider this code snippet unneccessary?

Reply:
Readded with updated snippet.

Reply:
Readded with updated snippet.